algorithms/pagerank.ddR/R/dpagerank.R

```diff
@@ -0,0 +1,280 @@
+
+
+  niter <- as.numeric(niter)
+  if (! niter > 0 )
+    stop("Invalid iteration count, it should be a positive number")
+  eps <- as.numeric(eps)
+  if (! eps > 0 )
+    stop("Invalid value for 'eps', it should be a positive number")
+  damping <- as.numeric(damping)
+  if (damping <= 0 || damping >= 1)
+    stop("Invalid damping factor, it should be between 0 and 1")
+
+  if (trace) {    # end of timing step
+    endtime <- proc.time()
+    spentTime <- endtime-starttime
+    cat("Spent time:",(spentTime)[3],"sec\n")
+  }
+
+  ### Initialization
+  if (trace) {
+    cat("Initialization PR and OutDegree\n")
+    starttime<-proc.time()
+  }
+  maxdiff <- eps
+  # initial values for PR is 1/nVertices like most literature (it is 1-damping in igraph package)
+  PR <- dmapply(function(data, nrow, ncol) matrix(data=data, nrow=nrow, ncol=ncol), 
+                ncol=psize(dgraph)[,2], MoreArgs=list(nrow=1, data=1-damping), output.type="darray", combine="cbind")
+
+  if (trace) {    # end of timing step
+    endtime <- proc.time()
+    spentTime <- endtime-starttime
+    cat("Spent time:",(spentTime)[3],"sec\n")
+  }
+  
+
+  #Let's get the number of outgoing edges in each partition
+  if (trace) {
+    cat("Calculating the number of outgoing edges\n")
+    starttime<-proc.time()
+  }
+
+  if (is.null(weights)) { # when there is no weight on the edges
+    OutDegree <- dmapply(function(X) X, rowSums(dgraph), output.type="darray", nparts=c(1, 1), combine='rbind') 
+  } else {  # when there are weights on the edges
+    weightedgraph <- dmapply(function(DGi, Wi) DGi * Wi, parts(dgraph), parts(weights), output.type=dgraph@type, nparts=nparts(dgraph), combine='rbind')
+    OutDegree <- dmapply(function(X) X, rowSums(weightedgraph), output.type="darray", nparts=c(1, 1), combine='rbind')
+  }
+
+  if (trace) {    # end of timing step
+    endtime <- proc.time()
+    spentTime <- endtime-starttime
+    cat("Spent time:",(spentTime)[3],"sec\n")
+  }
+
+
+  ### iterations
+  iteration_counter <- 1
+  if (trace)  iterations_start <- proc.time()
+  while (niter >= iteration_counter && maxdiff >= eps) {
+    if (trace) {
+      cat("Iteration: ",iteration_counter,"\n")
+      starttime <- proc.time()
+    }
+    iteration_counter <- iteration_counter + 1
+
+    # the new PR (PageRank vector) is calculated in each iteration
+    if(is.null(weights) && is.null(personalized)) {
+        PR_new <- dmapply(function(dg, prOld, damping, TPs, peri, wi) {
+                        if(class(dg) == "matrix") {
+                            .Call("pagerank_vm", prOld, dg, TPs, damping, NULL, NULL, PACKAGE="pagerank.ddR")
+                        } else {
+                            .Call("pagerank_spvm", prOld, dg, TPs, damping, NULL, NULL, PACKAGE="pagerank.ddR")
+                        }                        
+                    }, dg=parts(dgraph), 
+                    MoreArgs=list(prOld=PR, damping=damping, TPs=OutDegree), output.type="darray", combine="rbind")
```

Comment:
The default value of the nparts argument which is not specified in this dmapply will be c(1,totalParts(dgraph)). This indicates that the partitions are stitched together by cbind. This is inconsistent with what you have written in the combine which makes it seem that you want to rbind output partitions. Can you make sure this is what you have intended? Ideally you could change to combine = "cbind".

What is actually happening in the current code is that since nparts argument is not specified, the default value is the number of partitions of the input. Combine is only used when nparts < number of partitions of input.